eos-updater/apply.c

```diff
@@ -255,6 +334,15 @@ apply_internal (ApplyData     *apply_data,
                local_error->message);
   g_clear_error (&local_error);
 
+  /* Try to add a collection ID to the OS remote, which enables LAN/USB OS
+   * updates when combined with eos-updater.conf changes. Don't treat any
+   * errors as fatal to the update. */
+  /*FIXME: Remove this function call after the next checxpoint */
+  if (!update_remote_collection_id (repo, update_refspec, &local_error))
```

Comment:
I don't think so, because keeping them separate means we can easily drop this after the next checkpoint, and the performance difference between the two implementations is probably paltry.

Reply:
Not that it's hard to revert a patch, just that in terms of readable you can clearly tell this way that the two operations are separate

Reply:
I wasn’t thinking from a performance point of view, more from a ‘doing edits atomically is good’ point of view. It’s a bit moot since if anything else is writing the `config` file at the same time as the updater we’re doomed anyway, regardless of whether we do one write/replace or two.